src/crawler.cls.php

```diff
@@ -247,6 +247,37 @@ public static function async_handler($manually_run = false)
 		self::start($manually_run);
 	}
 
+	/**
+	 * Check if crawler can run in the choosen time period
+	 *
+	 * @since 6.1
+	 */
+	public static  function _crawler_in_schedule_time()
+	{
+		$now = new DateTime();
+		$class_settings = self::cls();
+		$schedule_times = $class_settings->conf(Base::O_CRAWLER_SCHEDULE_TIME, '');
+		if($schedule_times!== ''){
+			$schedule_times = explode(',', $schedule_times);
+
+			foreach ($schedule_times as $time) {
+				if ($time !== '') {
+					$hours = explode('-', $time);
+					$start = new DateTime($hours[0] . ":00");
+					$end = new DateTime($hours[1] . ":00");
+					if ($now < $end && $now > $start) {
+						return true;
+					}
+				}
+			}
```

Comment:
This will need more robust parsing and error handling.
What happens for these values of `$time`?
```php
'10';
'1-2';
'100-200';
'1:00:00-2:00:00';
'-2:00';
'00:00-';
'2:00-1:00'; // Should this be corrected or let it stay broken?
'1:00-1:00'; // Nearly impossible for the crawler to run; how to handle it?
'1:59-2:00'; // Valid, but does it cause problems?
// etc.
```

It looks like warnings will be logged in some of these cases. In others, an uncaught `Exception` is thrown.

Reply:
Added some parsing to limit incorrect time format